src/main/java/frisbee/puf/backend/network/ServerThread.java

```diff
@@ -43,27 +51,96 @@ public void run() {
         while (isRunning) {
             try {
                 //TODO: we need a shared object between client and server, like a request object
-                //Read request from client
-                String direction = (String) inFromClient.readObject();
-                log.info("Client:" + direction);
-
                 // TODO: when we have a shared object, we need to differentiate between the characters
                 // TODO: not sure how to do it with different threads for each client though
                 // TODO: the connection somehow needs to be made over the team
-                if (direction.equals("left") || direction.equals("right")) {
-                    //Send request to client
-                    log.info("To Client:" + direction);
-                    // TODO: return real value, for now just return the opposite of the own character value
-                    outToClient.writeObject(
-                            direction.equals("left") ? "right" : "left"
-                    );
+
+                String receivedJsonString = (String) inFromClient.readObject();
+
+                ObjectMapper objectMapper = new ObjectMapper();
+                SocketRequest clientRequest = objectMapper.readValue(receivedJsonString, new TypeReference<>() {
+                });
+
+                SocketRequestType messageType = clientRequest.getRequestType();
+
+                switch (messageType){
+                    case INIT:
+                        initClient(clientRequest);
+                        break;
+                    case MOVE:
+                        moveCharacter(clientRequest);
+                        break;
+                    case THROW:
+                        throwFrisbee(clientRequest);
+                        break;
+                    case GAME_RUNNING:
+                        startGame(clientRequest);
+                        break;
                 }
+
             } catch(Exception e){
                 e.printStackTrace();
                 isRunning = false;
+                SocketRequest response = new SocketRequest(SocketRequestType.READY, "false");
+                sendToClient(response);
             }
          }
     }
 
+    @SneakyThrows
+    private void initClient(SocketRequest clientRequest){
+        log.info(client.getInetAddress().getLocalHost() + "has sent a message of type INIT.");
+
+        this.teamName = clientRequest.getValue();
+        //add client to list of client threads
+        SocketServer.addClient(this.teamName, this);
+        Set<ServerThread> teamClientThreads = SocketServer.getClientThreadsByTeam().get(this.teamName);
+        boolean canStartGame = false;
+        if (teamClientThreads.size() == 2) {
+            canStartGame = true;
```

Comment:
Macht es Sinn, das vllt erst nach `if (!clientThread.equals(this)) {` zu setzen?

Reply:
Ja, ist logischer.